lib/dynamoid/type_casting.rb

```diff
@@ -288,8 +288,8 @@ def process(value)
 
     class BinaryTypeCaster < Base
       def process(value)
-        if value.is_a? String
-          value.dup
+        if value.is_a?(StringIO) || value.is_a?(IO)
+          value
         else
           value.to_s
         end
```

Comment:
Not sure if it's correct to remove the `#dup` call. Almost all parameters are duplicated at casting. Don't remember the reason and cannot find quickly a relevant GibHub issue. But if duplication should be removed - it should be done for all the field types in this file.

Reply:
Yeah, I noticed that too. Let me add it back just to be on the safe side. I'm not familiar with Dynamoid, just playing with it in the last couple days.